File: src/main/java/com/wanted/preonboarding/ticket/presentation/dto/PerformanceResponse.java
package com.wanted.preonboarding.ticket.presentation.dto;

import com.wanted.preonboarding.ticket.domain.Performance;
import com.wanted.preonboarding.ticket.domain.Showing;
import java.time.LocalDate;
import java.util.UUID;

public record PerformanceResponse(
	UUID performanceId,
	String performanceName,
	String performanceType,
	LocalDate startDate,
	boolean isReserve,
	int round

) {

	public static PerformanceResponse of(Showing entity) {
		Performance performance = entity.getPerformance();
		return new PerformanceResponse(
			entity.getId(),
			performance.getName(),
			performance.getType().toString(),
			entity.getStartDate(),
			entity.isReservationAvailable(),
			entity.getRound()
		);
	}

}

File: src/main/kotlin/com/wanted/preonboarding/ticket/presentation/response/PerformanceResponse.kt
package com.wanted.preonboarding.ticket.presentation.response

import com.wanted.preonboarding.ticket.domain.Performance
import java.time.LocalDateTime
import java.util.UUID

data class PerformanceResponse(
    val performanceId: UUID,
    val name: String,
    val round: Int,
    val startDate: LocalDateTime,
    val isReserve: Boolean,
) {
    companion object {
        fun from(performance: Performance): PerformanceResponse {
            return PerformanceResponse(
                performanceId = performance.id.value,
                name = performance.name,
                round = performance.round,
                startDate = performance.startDate,
                isReserve = performance.isReserveAvailable(),
            )
        }
    }
}
